### PR TITLE
CTA Swizzles

### PR DESCRIPTION
**Pull request by mmigdal-nv**
Cherry-pick of the changes made in https://github.com/NVIDIA/Fuser/pull/87 into main.

### CODE REVIEW

File: csrc/scheduler/matmul.cpp [hunk 1]
```diff
@@ -181,6 +181,29 @@ void scheduleMatmul(
   // [... M,N,K]
   scheduler_utils::matmul_utils::makeTile(cc, gemm_tile.cta_tile.toVector());
 
+  // Applies swizzle factor on C
+  if (params.grid_swizzle_factor != 1) {
+    int factor = std::max(1, params.grid_swizzle_factor); // must be >=1
```
Comment (csarofeen) [inline csrc/scheduler/matmul.cpp:186]: We don't consider this type of tiling to be a swizzle. We consider swizzles to be non affine transformations. Can we just get a different word for this optimization? Tile shuffle or something similar might be a good name.

**Review by csarofeen — APPROVED** [review]
This seems good to me, we should just be able to merge it in? CC @zasdfgbnm

**Comment by drzejan2** [comment]
@mmigdal-nv thanks for heads up, when this is promoted to `main` I will update #23 to incorporate this functionality in matmul scheduler in segmenter.

File: csrc/scheduler/matmul.cpp (hunk 1 above)
Comment (zasdfgbnm) [reply csrc/scheduler/matmul.cpp:186]: We are already using the word `swizzle` for both affine and non-affine transformations. See note `[WarpMmaSwizzler]`.

**Review by zasdfgbnm — APPROVED** [review]

File: csrc/scheduler/matmul.cpp (hunk 1 above)
Comment (mmigdal-nv) [reply csrc/scheduler/matmul.cpp:186]: I went with CUTLASS's naming. What about grid scaling ? Or I'm fine with tile shuffle. So should I update @zasdfgbnm @csarofeen ?

Comment (csarofeen) [reply csrc/scheduler/matmul.cpp:186]: No it's fine then, we should probably have a non-affine vs affine swizzle rename, but doesn't seem to make a difference right now.

Comment (zasdfgbnm) [reply csrc/scheduler/matmul.cpp:186]: Let's merge this PR as is so it is consistent with tracking-matmul.

**Pull request merged**